File: 6.1 HttpServer/httpserver2.js
// 2. Done

'use strict'; /* jshint
node : true
*/

var http = require('http'),
    url = require('url'),
    hello2 = function (req, res) {
        var uri = url.parse(req.url).pathname;
        res.writeHead(200, {'Content-Type' : 'text/html'});
        if (uri.substring(0,7) === '/hello/') {
            res.end('<h1>Node.js : Hello, ' + uri.replace('/hello/', '') + '!</h1>');
        }
        else {
            res.end('<h1>Node.js : Hello, world!</h1>');
        }
    };

http.createServer(hello2).listen(8888);

File: 6.1 HttpServer/httpserver2.go
// 2. Done

package main

import (
	"io"
	"net/http"
)

func hello1(w http.ResponseWriter, r *http.Request) {
	io.WriteString(w, "<h1>Go : Hello, world!</h1>")
}

func hello2(w http.ResponseWriter, r *http.Request) {
	uri := r.URL.Path[7:] // Ignore '/hello/'
	io.WriteString(w, "<h1>Go : Hello "+uri+"!</h1>")
}

func main() {
	http.HandleFunc("/", hello1)
	http.HandleFunc("/hello/", hello2)
	http.ListenAndServe(":8888", nil)
}
